### PR TITLE
Add --fix-remotes option

### PR DESCRIPTION
**Pull request by cvtsi2sd**
implemented only in the git case to switch remote if it doesn't match instead of throwing a fit; useful if a repo from the catalog changed URL

### CODE REVIEW

File: quark/subproject.py [hunk 1]
```diff
@@ -198,7 +198,7 @@ def add_module(parent, name, uri, options, conf, **kwargs):
         while len(stack):
             current_module = stack.pop()
             if current_module.external_project:
-                generate_cmake_script(current_module.directory, update = update)
+                generate_cmake_script(current_module.directory, update = update, clean = clean, clobber = clobber, fix_remotes=fix_remotes)
```
Comment (mbeniamino) [inline quark/subproject.py:201]: I see you added the previously missing `clean` and `clobber` parameters. Logically it seems correct to me, but is this a fix for a previous bug? If so, it should go in a separate commit.

File: quark/subproject.py [hunk 2]
```diff
@@ -348,26 +348,25 @@ def checkout(self):
...
-but its remote 'origin' (%r)
-does not match what we expect (%r).
-
-Please either remove the local clone, or fix its remote.""" % (self.directory, current_origin, self.url.geturl()))
+                    if fix_remotes:
+                        logger.info("Fixing incorrect remote in %s directory (%s -> %s)" % (self.directory, current_origin, self.url.geturl()))
+                        fork(['git', 'remote', 'set-url', 'origin', self.url.geturl()])
+                    else:
+                        raise QuarkError("""
+
```
Comment (mbeniamino) [inline quark/subproject.py:364]: Is there a reason why now this and the remaining lines in the string are indented?

File: quark/subproject.py (hunk 1 above)
Comment (cvtsi2sd) [reply quark/subproject.py:201]: Yeah, it's essentially a bugfix; I thought about this, but I was lazy. I'll split it out now.

File: quark/subproject.py (hunk 2 above)
Comment (cvtsi2sd) [reply quark/subproject.py:364]: Right, I blindly indented everything when moving under the `else` branch, but that's a `"""` string. Fixing...

**Comment by cvtsi2sd** [comment]
Fixed/squashed/...
